packages/socket-mode/src/SocketModeClient.ts

```diff
@@ -398,6 +413,37 @@ export class SocketModeClient extends EventEmitter {
       }
     });
   }
+
+  /**
+   * Removes secrets and tokens from socket request and response objects
+   * before logging.
+   * @param body - the object with values for redaction.
+   * @returns the same object with redacted values.
+   */
+  private static redact(body: NestedRecord): NestedRecord {
+    if (body === undefined) {
+      return body;
+    }
+    const record = Object.create(body);
+    if (Array.isArray(body)) {
+      return body.map((item) => (
+        (typeof item === 'object' && item !== null) ?
+          SocketModeClient.redact(item) :
+          item
+      ));
+    }
+    Object.keys(body).forEach((key: string) => {
+      const value = body[key];
+      if (typeof value === 'object' && value !== null) {
+        record[key] = SocketModeClient.redact(value as NestedRecord);
```

Comment:
This cast seems needed even with the recursive definition above... 🤔 I fumbled a bit with it without too much luck.

```
Argument of type 'object' is not assignable to parameter of type 'NestedRecord'.
```